pkg/scraper/scraper.go

```diff
@@ -18,46 +18,58 @@ const (
 	Bruneida  = 2
 )
 
-type scraper struct {
-	wg   sync.WaitGroup
-	jobs []*model.Job
-}
+type (
+	Scraper interface {
+		// ScrapeJobs scrapes provider's website for job listings and company information if available.
+		// Returned Job struct contains a Company struct with ProviderCompanyID only.
+		// Company details are returned as a separate argument.
+		ScrapeJobs() ([]*model.Job, error)
+	}
 
-func createScraper() scraper {
-	return scraper{wg: sync.WaitGroup{}, jobs: []*model.Job{}}
-}
+	JobCentreScraper struct {
+		BaseURL     string
+		FetchClient FetchClienter
+	}
+
+	BruneidaScraper struct {
+		BaseURL     string
+		FetchClient FetchClienter
+	}
+
+	FetchClienter interface {
+		GetDocument(url string) (*goquery.Document, error)
+	}
+
+	FetchClient struct{}
+)
 
-func getDocument(url string) (*goquery.Document, error) {
-	fmt.Printf("Visiting: %s \n", url)
+func (c *FetchClient) GetDocument(url string) (*goquery.Document, error) {
+	log.Debug().Msgf("Visiting: %s \n", url)
 	var doc *goquery.Document
 
 	err := retry.Do(func() error {
-
 		res, err := http.Get(url)
 		if err != nil {
-			return err
+			return errors.Wrapf(err, "Error in HTTP GET")
 		}
 
 		defer res.Body.Close()
 
 		if res.StatusCode != http.StatusOK {
 			errorMessage := fmt.Sprintf("status code error: %d %s", res.StatusCode, res.Status)
 
-			fmt.Println(errorMessage)
-
-			return errors.New(errorMessage)
+			return fmt.Errorf("Status is not 200 OK: %s", errorMessage)
```

Comment:
why this line has different method of returning error? Why don't we use `errors.Wrapf`?

Reply:
It is because there is no `err` here. A non-successful HTTP status code does not result in an error. It is up to us to return an error if we only want 200 OK to be successful.